File: external-libs/core/src/main/java/com/example/core/security/impl/crypto/Result.java
package com.example.core.security.impl.crypto;

public class Result {

    private String message;
    private byte[] iv;
    private CryptoError error;

    public Result(String message, byte[] iv, CryptoError error) {
        this.message = message;
        this.iv = iv;
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public byte[] getIv() {
        return iv;
    }

    public CryptoError getError() {
        return error;
    }

    public boolean isResultAvailable() {
        return error == CryptoError.OK;
    }

    @Override
    public String toString() {
        return message;
    }
}

File: multiplatform/Notes/composeApp/src/androidMain/kotlin/com/notes/app/security/Result.kt
package com.notes.app.security

import com.notes.app.security.CryptoKeystore.CryptoError

data class Result(
    // Can contain IV data part
    val message: String = "",
    val iv: String = "",
    val error: CryptoError = CryptoError.UNKNOWN,
    val hasIV: Boolean = false,
) {
    // A message without IV data part
    val realMessage =
        if (hasIV && message.isNotEmpty()) {
            message.substring(CryptoKeystore.IV_SIZE)
        } else {
            message
        }

    val success: Boolean
        get() = error === CryptoError.OK

    override fun toString(): String = ""
}
